### PR TITLE
OPENEUROPA-777: Make sure component dependencies are as relaxed as possible.

### PR DESCRIPTION
**Pull request by aritomelo**
Set Minimal minor version.

And also added decimal numer to keep the consistency.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -33,3 +33,6 @@ matrix:
   IMAGE_PHP:
     - fpfis/httpd-php-dev:5.6
     - fpfis/httpd-php-dev:7.1
+  COMPOSER_BOUNDARY:
```
Comment (pfrenssen) [inline .drone.yml]: The matrix is extended with choices for the composer boundary, but this `${COMPOSER_BOUNDARY}` variable is not used in the pipeline, so it has no effect.

For an example on how to use this, see the pipeline of the Code Review component: https://github.com/openeuropa/code-review/blob/master/.drone.yml#L12

File: composer.json [hunk 2]
```diff
@@ -6,11 +6,11 @@
   "minimum-stability": "dev",
   "prefer-stable": true,
   "require": {
-    "twig/twig": "~1"
+    "twig/twig": "~1.0"
   },
   "require-dev": {
-    "phpunit/phpunit": "~5||~6.0",
-    "openeuropa/code-review": "~0.3"
+    "phpunit/phpunit": "~5.0||~6.0",
```
Comment (pfrenssen) [inline composer.json]: Let's also update this to remove the double pipe, this is not according to the Composer specification. Let's make it `~5.0|~6.0`.

File: .drone.yml (hunk 1 above)
Comment (voidtek) [reply .drone.yml]: 👍

Comment (voidtek) [reply .drone.yml]: I am not confortable with the syntax proposed in code-review
```
composer-install-lowest:
    group: prepare
    image: ${IMAGE_PHP=fpfis/httpd-php-dev:5.6}
    volumes:
      - /cache:/cache
    commands:
      - composer update --prefer-lowest
    when:
      matrix:
        COMPOSER_BOUNDARY: lowest
   composer-install-highest:
    group: prepare
    image: ${IMAGE_PHP=fpfis/httpd-php-dev:5.6}
    volumes:
      - /cache:/cache
    commands:
      - composer update
    when:
      matrix:
        COMPOSER_BOUNDARY: highest
[...]
  COMPOSER_BOUNDARY:
    - lowest
    - highest
```

I prefer the following syntax:
```
composer-install:
    group: prepare
    image: ${IMAGE_PHP=fpfis/httpd-php-dev:5.6}
    volumes:
      - /cache:/cache
    commands:
      - composer update ${COMPOSER_BOUNDARY}
[...]
  COMPOSER_BOUNDARY:
    - '--prefer-lowest'
    - ''
```

**Comment by voidtek** [comment]
@richardcanoe,  I revert your commit according to my comment https://github.com/openeuropa/ecl-twig-loader/pull/9#discussion_r216946476

File: .drone.yml (hunk 1 above)
Comment (dxvargas) [reply .drone.yml]: It's better like this, I approve @voidtek .

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# @see http://editorconfig.org/
+
+# This is the top-most .editorconfig file; do not search in parent directories.
+root = true
+
+# All files.
+[*]
+end_of_line = LF
+indent_style = space
+indent_size = 4
```
Comment (pfrenssen) [inline .editorconfig]: I am not sure whether these options are correct for this project. Is the indent style really set to 4 for all the files in the project? I see that the `.drone.yml` file for example is using an indent of 2 spaces. This is in conflict with what is being proclaimed here.

I would suggest to take this out of this PR for now, it is out of scope and seems to need additional work.

**Comment by pfrenssen** [comment]
This is looking good now, but one of my earlier remarks turned out to be incorrect as was pointed out by @dxvargas in one of the sister issues. I suggested to change the "or" clause in the PHPUnit version string from a double pipe to a single pipe, but in fact the double pipe is the preferred symbol, as documented in [Versions and constraints - version range](https://getcomposer.org/doc/articles/versions.md#version-range). The single pipe is only supported for backwards compatibility reasons.

I will make this small correction and approve the PR.

**Pull request merged**